### PR TITLE
[4][com_finder] error when gather statistic enabled

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #38370 .

### CODE REVIEW

**Comment by Kostelano** [comment]
Yes, it works for [my error](https://github.com/joomla/joomla-cms/issues/38370#issue-1325456406) too.

**Comment by Kostelano** [comment]
Will the PR be marked as ready? Or are you looking for another way to solve the problem?

**Comment by alikon** [comment]
i really would like to hear from the _**thumbs down superstar**_, maybe he/she/it could explain what's wrong here and maybe give us a proper fix
anyway i'm digressing
that's my quick & dirty fix,
if you don't like it @SharkyKZ please propose yours

**Comment by brianteeman** [comment]
You are removing the sanitization on the query

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on eb8aa81539492bdf7840b387f8ff4053fbc7184e<br /><br />I'm posting a successful test as it works. Even if at the moment there is no better solution.

This needs to be fixed, otherwise users using the "Gather Search Statistics" option will be pleasantly surprised. I don't know how a release blocker is defined, but it looks like it.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38373">issues.joomla.org/tracker/joomla-cms/38373</a>.</sub>

**Comment by Kostelano** [comment]
It is strange that they did not pay attention to this. In addition to @WeibelMarc's appeal, there is another one on the [Russian-language forum](https://joomlaforum.ru/index.php/topic,358468.msg1834552.html#msg1834552).

File: components/com_finder/src/Helper/FinderHelper.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static function logSearch(Query $searchquery, $resultCount = 0)
         $query = $db->getQuery(true);
 
         // Sanitise the term for the database
-        $temp = unserialize(serialize($searchquery));
+        $temp = new \stdClass();
         $temp->input = trim(strtolower($searchquery->input));
```
Comment (roland-d) [inline components/com_finder/src/Helper/FinderHelper.php]: @alikon To bring sanitation back I think we can add `(string)` before the `$searchquery->input` so the line reads:

`$temp->input = trim(strtolower((string) $searchquery->input));`

**Review by roland-d — CHANGES_REQUESTED** [review]

File: components/com_finder/src/Helper/FinderHelper.php (hunk 1 above)
Comment (alikon) [reply components/com_finder/src/Helper/FinderHelper.php]: done

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 11fbfbd0b7c97c493cdf21b0177156cc5911d31a<br /><br />After the changes everything works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38373">issues.joomla.org/tracker/joomla-cms/38373</a>.</sub>

**Comment by bayareajenn** [comment]
I have tested this item :white_check_mark: successfully on 11fbfbd0b7c97c493cdf21b0177156cc5911d31a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38373">issues.joomla.org/tracker/joomla-cms/38373</a>.</sub>

**Review by roland-d — APPROVED** [review]

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**